File: IT/2/Database/Esercizi/Python/esercizio3.py
# file esercizio3.py

from __future__ import print_function
import sqlite3

conn = sqlite3.connect("students.db")
cursor = conn.cursor()

try:
	cursor.execute("SELECT number, name, surname FROM students")
	res = cursor.fetchall()

	if not res:
		print("Non ci sono studenti iscritti al corso!")
		exit()

	for student in res:
		print("Matricola {}, {} {}.".format(student[0], student[1], student[2]))

except sqlite3.OperationalError as e:
	print(e)

File: IT/2/Database/Esercizi/C++/esercizio3.cpp
/* file esercizio3.cpp */

#include <iostream>
#include <sqlite3.h>

using std::string;
using std::cout;
using std::endl;

int main()
{
	sqlite3 *db;
	sqlite3_stmt *res;
	sqlite3_open("students.db", &db);

	int rc = sqlite3_prepare_v2(db, "SELECT number, name, surname FROM students", -1, &res, NULL);

	if (rc != SQLITE_ERROR) {
		if (sqlite3_step(res) == SQLITE_DONE) {
			cout << "Non ci sono studenti iscritti al corso!" << endl;
			sqlite3_finalize(res);
			sqlite3_close(db);
			return 0;
		}

		do {
			string number = (const char*) sqlite3_column_text(res, 0);
			string name = (const char*) sqlite3_column_text(res, 1);
			string surname = (const char*) sqlite3_column_text(res, 2);
			cout << "Matricola " << number << ", " << name << " " << surname << "." << endl;
		}
		while (sqlite3_step(res) == SQLITE_ROW);
	}
	else {
		cout << sqlite3_errmsg(db) << endl;
	}

	sqlite3_finalize(res);
	sqlite3_close(db);
}

